### PR TITLE
Support text selection programmatically

### LINKED ISSUE #69460: Support text selection programmatically for the SelectableText widget
Flutter currently does not provide a way to select text programmatically for the SelectableText widget. 

On Android we do that using the `Spannable` API, see https://stackoverflow.com/a/41760489/9185192
On iOS we a property to define a selection range, see https://developer.apple.com/documentation/uikit/uitextview/1618615-selectedrange

### Proposal

It would be great to have this capability in Flutter to help native developers easily migrate their apps to Flutter. I suggest to have this capability on `SelectableText` widget since it is the only one text visualizer that support text selection.

### PR DESCRIPTION
**Pull request by pedromassango**
## Description

This PR add support for text selection programmatically for the `SelectableText` widget.
See [design doc](https://docs.google.com/document/d/1NK4GJ0Wk5eJ3IUw_9oAtz1oYgZ-Sn-kAKHxTvsgr75o/edit?usp=sharing).

Note: this is a continuation of https://github.com/flutter/flutter/pull/69461.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/69460

## Tests

TODO

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

**Review by chunhtai — COMMENTED** [review]
Since I am working on a more generic selection interface, and It will be moving away from TextEditingController. I think we should hold off this change.

File: packages/flutter/lib/src/material/selectable_text.dart [hunk 1]
```diff
@@ -511,6 +538,16 @@ class _SelectableTextState extends State<SelectableText> implements TextSelectio
     super.dispose();
   }
 
+  void _updateControllerAndSubscribe() {
+    _controller = _TextSpanEditingController(
+        textSpan: widget.textSpan ?? TextSpan(text: widget.data)
+    );
+    if (widget.controller != null) {
+      widget.controller!._textSpanEditingController = _controller;
```
[the remark this answers is not in the document]
Comment (pedromassango) [reply packages/flutter/lib/src/material/selectable_text.dart:546]: Interesting, I was just afraid to refactor that code

**Comment by Hixie** [comment]
Since there's a more elaborate change coming in this space, it sounds like we should close this PR. I'm going to do so, please don't hesitate to reopen if I was mistaken! I recommend following up in the #hackers-text-input channel on Discord.

**Comment by wanghongenpin** [comment]
> Since I am working on a more generic selection interface, and It will be moving away from TextEditingController. I think we should hold off this change.因为我正在开发一个更通用的选择界面，它将远离TextEditingController。我想我们应该推迟这个变化。

How is the current progress

**Pull request closed without merging**